src/toolkits/nearest_neighbors/nearest_neighbors.cpp

```diff
@@ -318,6 +318,25 @@ void nearest_neighbors_model::train(const sframe& X,
   train(X, ref_labels, composite_distance_params, opts);
 }
 
+void nearest_neighbors_model::populate_distance_for_summary_struct(
+    const std::vector<dist_component_type>& composite_distance_params) {
+  if (composite_distance_params.size() > 1) {
```

Comment:
I'm struggling to understand what each of these three cases are for. I think a few short comments would really help.

Reply:
Done.